src/module.c

```diff
@@ -6498,7 +6586,15 @@ RedisModuleCallReply *RM_Call(RedisModuleCtx *ctx, const char *cmdname, const ch
 cleanup:
     if (reply) autoMemoryAdd(ctx,REDISMODULE_AM_REPLY,reply);
     if (ctx->module) ctx->module->in_call--;
-    if (c) moduleReleaseTempClient(c);
+    if (!ctx->persistent_client) {
```

Comment:
Maybe we should add a new flag to mark the persisted module client. This way we can just check the flag here on `c`. Otherwise I think it is confusing that we check some value on the context and then perform free the argv on `c`.